tm2/pkg/commands/ffcli/README.md

```diff
@@ -0,0 +1,153 @@
+# ffcli [![go.dev reference](https://img.shields.io/badge/go.dev-reference-007d9c?logo=go&logoColor=white&style=flat-square)](https://pkg.go.dev/github.com/peterbourgon/ff/v3/ffcli)
+
+ffcli stands for flags-first command line interface,
+and provides an opinionated way to build CLIs.
+
+## Rationale
+
+Popular CLI frameworks like [spf13/cobra][cobra], [urfave/cli][urfave], or
+[alecthomas/kingpin][kingpin] tend to have extremely large APIs, to support a
+large number of "table stakes" features.
+
+[cobra]: https://github.com/spf13/cobra
+[urfave]: https://github.com/urfave/cli
+[kingpin]: https://github.com/alecthomas/kingpin
+
+This package is intended to be a lightweight alternative to those packages. In
+contrast to them, the API surface area of package ffcli is very small, with the
+immediate goal of being intuitive and productive, and the long-term goal of
+supporting commandline applications that are substantially easier to understand
+and maintain.
+
+To support these goals, the package is concerned only with the core mechanics of
+defining a command tree, parsing flags, and selecting a command to run. It does
+not intend to be a one-stop-shop for everything your commandline application
+needs. Features like tab completion or colorized output are orthogonal to
+command tree parsing, and should be easy to provide on top of ffcli.
+
+Finally, this package follows in the philosophy of its parent package ff, or
+"flags-first". Flags, and more specifically the Go stdlib flag.FlagSet, should
+be the primary mechanism of getting configuration from the execution environment
+into your program. The affordances provided by package ff, including environment
+variable and config file parsing, are also available in package ffcli. Support
+for other flag packages is a non-goal.
+
+
+## Goals
+
+- Absolute minimum usable API
+- Prefer using existing language features/patterns/abstractions whenever possible
+- Enable integration-style testing of CLIs with mockable dependencies
+- No global state
+
+## Non-goals
+
+- All conceivably useful features
+- Integration with flag packages other than [package flag][flag] and [ff][ff]
+
+[flag]: https://golang.org/pkg/flag
+[ff]: https://github.com/peterbourgon/ff
+
+## Usage
```

Comment:
Not sure if we should start partially migrating ffcli, or if we should just reimplement it our way and give credits.

Reply:
(and keep only `peterbourgon/ff` for env/config overriding)

Reply:
Would you prefer to start the re-implementation process now, or do you recommend merging this intermediary step first? Could you explain your reasoning?

(I prefer reimplementing now)

Reply:
I prefer to reimplement because the current change is a little half-baked. I did it to facilitate the review, but it's OK for you, let's reimplement fully now.

Reply:
I converted the PR to draft.

Reply:
@moul I merged ffcli code in `pkg/commands`, thanks for restarting the review.